File: java/src/test/java/fun/pierre/lists/P04Test.java
package fun.pierre.lists;

import org.junit.Test;
import static java.util.Arrays.asList;
import static org.hamcrest.CoreMatchers.equalTo;
import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertThat;

public class P04Test {
    @Test
    public void shouldFindNumberElementOfIntList() {
        assertThat(P04.numbeElements(asList(1, 2, 3, 4, 5)), is(equalTo(5)));
    }

    @Test
    public void shouldFindNunberThElementOfCharList() {
        assertThat(P04.numbeElements(asList('a', 'b', 'c', 'd')), is(equalTo(4)));
    }
}

File: scala/src/test/scala/fun/pierre/lists/P04Test.scala
package fun.pierre.lists

import org.scalatest.{FunSpec, Matchers}

class P04Test extends FunSpec with Matchers {

  describe("P04 Number element") {
    it("P04 number char element") {
      val k = P04.numberElement(List("a", "b", "c", "d"))
      k should be(4)
    }

    it("P04 number int element") {
      val k = P04.numberElement(List(1, 2, 3, 4, 5))
      k should be(5)
    }
  }
}
